bundles/AdminBundle/Resources/public/js/pimcore/settings/recyclebin.js

```diff
@@ -187,34 +192,72 @@ pimcore.settings.recyclebin = Class.create({
     },
 
     updateButtonStates: function() {
-        var selectedRows = this.grid.getSelectionModel().getSelection();
+        let selectedRows = this.grid.getSelectionModel().getSelection();
+        let selectedCount = selectedRows.length;
+        let deleteButton = true;
+        let publishButton = true;
 
-        if (selectedRows.length >= 1) {
-            Ext.getCmp("pimcore_recyclebin_button_restore").enable();
+        Ext.getCmp("pimcore_recyclebin_button_restore").disable();
+        Ext.getCmp("pimcore_recyclebin_button_delete").disable();
+
+        for (let i = 0; i < selectedCount; i++) {
+            //it only takes one element not deletable to disable the delete button
+            if (selectedRows[i].data.permissions.delete == 0) {
+                //it only takes one not deleteable element among selected to disable the delete button
+                deleteButton = false;
+                break;
+            }
+        }
+        if (deleteButton && selectedCount > 0){
             Ext.getCmp("pimcore_recyclebin_button_delete").enable();
-        } else {
-            Ext.getCmp("pimcore_recyclebin_button_restore").disable();
-            Ext.getCmp("pimcore_recyclebin_button_delete").disable();
+        }
+
+        for (let i = 0; i < selectedCount; i++) {
+            //it only takes one not publishable element among selected to disable the publish button
+            if (selectedRows[i].data.permissions.publish == 0) {
+                publishButton = false;
+                break;
+            }
+        }
+        if (publishButton && selectedCount > 0){
+            Ext.getCmp("pimcore_recyclebin_button_restore").enable();
```

Comment:
can i still restore the element if i have `list` & `view` permission and not `save` or `publish`?

Reply:
`restore` is only available for `publish` permission, see https://github.com/pimcore/pimcore/blob/bdc9a2dcd6d75403a3a1750c4c117fc8e1ea541e/models/Element/Recyclebin/Item.php#L132

The idea is that after PR, we can, for whatever reason and use case, define user/roles that only can see and list the recycle bin and not necessary can perform any action like restore or flush.